### PR TITLE
Update dependencies, add category ID and href to facets response

### PR DESCRIPTION
**Pull request by chrsmutti**
#### What problem is this solving?

Add Category ID and HREF to facets response. I've also updated our dependencies, linting, and formatting some typescript 3.7.x features were not being done correctly because of the version of `eslint` and `prettier`.

I've also added here a small workaround for accounts with languages other than English, we're only checking for the attribute key `price`, but it changes based on the binding locale. I've added some common ones, this workaround can be dropped once we finish the facets rework that is scheduled for this quarter.

#### How should this be manually tested?

[EriksBikeShop](https://christian--eriksbikeshop.myvtex.com/)
[Umbro](https://christian--umbro.myvtex.com/)
[Biscoind](https://christian--biscoind.myvtex.com/)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [x] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage

image
image

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes

Some of the changes made were done following the conventions on the linting guide, now that it's working again, there are a lot of linter warnings and errors to be fixed, I've tried fixing only on files this PR would modify either way and then we can in the future create another one specifically to address linting issues.

### CODE REVIEW

File: node/utils/attributes.ts [hunk 1]
```diff
@@ -186,22 +170,36 @@ const convertValues = (
...
+): string => {
+  if (isEmpty(key) || isEmpty(value)) {
+    return baseHref
+  }
+
+  const [path = '', map = ''] = baseHref.split('?map=')
+  const pathValues = [...path.split('/'), value].filter((x) => !isEmpty(x))
+  const mapValues = [...map.split(','), key].filter((x) => !isEmpty(x))
+
+  return `${pathValues.join('/')}?map=${mapValues.join(',')}`
```
Comment (LucasTornai) [inline node/utils/attributes.ts:186]: I suppose there's no way for it to happen (?), but you never check whether pathValues and mapValues are both not empty at the same time

**Review by LucasTornai — APPROVED** [review]

File: node/utils/attributes.ts (hunk 1 above)
Comment (chrsmutti) [reply node/utils/attributes.ts:186]: I guess this test case would cover this issue, wouldn't it?
https://github.com/vtex-apps/search-resolver/pull/64/files/df5d988a5231c20a6760610d069aecb1dfe852eb#diff-4ec3b647f6eda70d0edb78e6069561caR285

As it's always adding `key` to the end of the `pathValues` array, and value to the end of `mapValues` array, the arrays themselves cannot be empty if `key` and `value` exists. The check for `key` and `value` is on line 178. Does this cover?

Comment (LucasTornai) [reply node/utils/attributes.ts:186]: Yeah, I guess this covers it

**Review by hiagolcm — APPROVED** [review]

**Review by thalytafabrine — COMMENTED** [review]
do you know why some categories still have `id` and `href` with `null` value?

image

**Comment by chrsmutti** [comment]
> do you know why some categories still have `id` and `href` with `null` value?

Category IDs being `null` is due to some products that haven't been indexed still by the changes made in the backend, recently it was triggered a full reindex of all account's catalog so that those fields are correctly set, but as `null` was the old behavior, I don't see a problem in it being temporarily that way.

The `href` is a bit strange, I can't think of a situation it would not be set, as it's dynamically set on the resolver, do you have the workspace and query you used @thalytafabrine? I'll take a look.

File: node/commons/compatibility-layer.ts [hunk 2]
```diff
@@ -14,27 +10,35 @@ export const convertBiggyProduct = (
...
+        })
+        .reverse()
+    : []
 
   const skus: SearchItem[] = (product.skus || []).map(
     convertSKU(product, indexingType, tradePolicy)
   )
 
-  const allSpecifications = product.productSpecifications.concat(getSKUSpecifications(product))
+  const productSpecifications = product.productSpecifications ?? []
```
Comment (chrsmutti) [inline node/commons/compatibility-layer.ts:32]: This was required as a workaround on some products having `productSpecifications` set as `undefined`, @LucasTornai is taking a look 😄

**Comment by thalytafabrine** [comment]
do you know why some categories still have id and href with null value?

> > do you know why some categories still have `id` and `href` with `null` value?
> 
> Category IDs being `null` is due to some products that haven't been indexed still by the changes made in the backend, recently it was triggered a full reindex of all account's catalog so that those fields are correctly set, but as `null` was the old behavior, I don't see a problem in it being temporarily that way.
> 
> The `href` is a bit strange, I can't think of a situation it would not be set, as it's dynamically set on the resolver, do you have the workspace and query you used @thalytafabrine? I'll take a look.

I tried on `eriksbikeshop`: https://christian--eriksbikeshop.myvtex.com/bike?_q=bike&map=ft

**Comment by chrsmutti** [comment]
> I tried on `eriksbikeshop`: https://christian--eriksbikeshop.myvtex.com/bike?_q=bike&map=ft

My bad 😅 I forgot i linked that workspace as a reference to this PR, and tested something else in the same workspace.
I've reinstalled the version with the changes mentioned here, and it seems to work as expected:

image

**Review by thalytafabrine — APPROVED** [review]

File: node/commons/compatibility-layer.ts (hunk 2 above)
Comment (LucasTornai) [reply node/commons/compatibility-layer.ts:32]: Just for the record, seems that it isn't affecting any other store for now, so I won't publish a fix beforehand and let this PR fix it.

**Comment by diyoda** [comment]
Hi @chrsmutti Is there a reason for this PR for not getting merged? 

We have a requirement for client SBD that will require this feature.

**Comment by chrsmutti** [comment]
> Hi @chrsmutti Is there a reason for this PR for not getting merged?
> 
> We have a requirement for client SBD that will require this feature.

Hi @diyoda, I'll look into rebasing the necessary changes and deploying the category ID and href response separately. Will let you know when the PR hits stable.

**Comment by chrsmutti** [comment]
Closed in favor of #112.

**Pull request closed without merging**